### PR TITLE
DRILL-8034: Add support for java 17

### PR DESCRIPTION
**Pull request by vvysotskyi**
# [DRILL-8034](https://issues.apache.org/jira/browse/DRILL-8034): Add support for java 17

## Description
- Update Java version to 17
- Update Github Actions to run tests for JDK 17 instead of 14
- Update Docker scripts to deploy build with 17

## Documentation
Yep

## Testing
All unit tests passes, checked that Drill is able to start on JDK 17 without errors or warnings.

### CODE REVIEW

File: distribution/src/main/resources/drill-config.sh [hunk 1]
```diff
@@ -307,10 +307,11 @@ export DRILLBIT_OPTS="$DRILLBIT_OPTS -XX:ReservedCodeCacheSize=$DRILLBIT_CODE_CA
 if [ $? -gt 0 ]; then
   # Allow reflective access on Java 9+
   export DRILLBIT_OPTS="$DRILLBIT_OPTS --add-opens java.base/java.lang=ALL-UNNAMED"
+  export DRILLBIT_OPTS="$DRILLBIT_OPTS --add-opens=java.base/java.util=ALL-UNNAMED"
   export DRILLBIT_OPTS="$DRILLBIT_OPTS --add-opens java.base/sun.nio.ch=ALL-UNNAMED"
+  export DRILLBIT_OPTS="$DRILLBIT_OPTS --add-opens java.base/java.net=ALL-UNNAMED"
```
Comment (vdiravka) [inline distribution/src/main/resources/drill-config.sh:312]: It is due to reflection limitations which comes starting from jdk16, right?

File: distribution/src/main/resources/drill-config.sh [hunk 2]
```diff
@@ -307,10 +307,11 @@ export DRILLBIT_OPTS="$DRILLBIT_OPTS -XX:ReservedCodeCacheSize=$DRILLBIT_CODE_CA
...
   export DRILLBIT_OPTS="$DRILLBIT_OPTS --add-opens java.base/java.nio=ALL-UNNAMED"
   export DRILLBIT_OPTS="$DRILLBIT_OPTS --add-opens java.security.jgss/sun.security.krb5=ALL-UNNAMED"
-  export DRILLBIT_OPTS="$DRILLBIT_OPTS --illegal-access=permit"
```
Comment (vdiravka) [inline distribution/src/main/resources/drill-config.sh:313]: Why did we have this config at all? :) It was for ASM?

File: pom.xml [hunk 3]
```diff
@@ -1015,6 +1017,11 @@
               <artifactId>asm</artifactId>
               <version>${asm.version}</version>
             </dependency>
+            <dependency>
```
Comment (vdiravka) [inline pom.xml:1020]: I got explanation offline, but pls do share here the reason for adding this dependency for `maven-shade-plugin`

File: pom.xml [hunk 4]
```diff
@@ -3965,6 +3972,22 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>jdk15</id>
```
Comment (vdiravka) [inline pom.xml:3976]: Is that for jdk15 only? If not possibly name it `<id>jdk9+</id>`

**Review by vdiravka — COMMENTED** [review]
Thanks for the following my [WIP]:(https://github.com/vdiravka/drill/commit/2bf1dacfd9b2b3a22749be2611d19c0042227bc0)

GJ. It wasn't comfortable to switch Java version for Maven for different projects.

Minor comments

File: pom.xml (hunk 4 above)
Comment (vvysotskyi) [reply pom.xml:3976]: Thanks, renamed.

**Review by vvysotskyi — COMMENTED** [review]
@vdiravka, thanks for the review, I've addressed your CR comments.

History note: this PR was made based on my previous attempt to update to JDK 15 in Feb this year: https://github.com/apache/drill/compare/39c4cc8482cabd36ada4e50effe68e0738ddbd67...27b219b8c9560b4a797a9b3a88164b1694fcc2ff
There were some strange failures, so I decided to postpone it, but resumed this work since the new LTS version was released and it became too popular to be ignored.

**Review by vdiravka — APPROVED** [review]
+1 LGTM
Thanks for adding JDK17!

**Pull request merged**